pkg/controller/vsphere/reconciler.go

```diff
@@ -633,6 +638,26 @@ func (vm *virtualMachine) getPowerState() (types.VirtualMachinePowerState, error
 	}
 }
 
+func (vm *virtualMachine) reconcileTags(ctx context.Context, session *session.Session, machine *machinev1.Machine) error {
```

Comment:
Should add a comment explaining what this does, eg
```suggestion
// reconcileTags ensures that the required tags are present on the virtual machine, eg the Cluster ID
func (vm *virtualMachine) reconcileTags(ctx context.Context, session *session.Session, machine *machinev1.Machine) error {
```

Reply:
👍